### DISCUSSION TITLE
Out of memory

### DISCUSSION THREAD

**Discussion by jiangzhuangxiansheng** [sebastienros/jint#1330] [Q&A]
![1666690554956](https://user-images.githubusercontent.com/33273148/197739724-94172f23-9568-4e87-9d9c-35d49d394cb9.jpg)
I used Transient mode to register about 50 times per second, and the memory kept growing. Should I change to Singleton mode for registration

**Comment by lahma** [comment]
Well if it helps? Engine is not thread-safe so if you make it a singleton, you need to synchronize access to it.
